tests/integration/support/steps.go

```diff
@@ -343,6 +365,51 @@ func (c *ScenarioCtx) PayloadsOfDispatchedEventsDontIncludeDecisions() error {
 	return nil
 }
 
+// TheUserProfileServiceStateShouldBe checks current state of UPS
+func (c *ScenarioCtx) TheUserProfileServiceStateShouldBe(value *gherkin.DocString) error {
+	config, err := c.clientWrapper.Client.GetProjectConfig()
+	if err != nil {
+		return fmt.Errorf("Invalid Project Config")
+	}
+	rawProfiles, err := parseYamlArray(value.Content, config)
+	if err != nil {
+		return fmt.Errorf("Invalid request for user profile service state")
+	}
+
+	expectedProfiles := userprofileservice.ParseUserProfiles(rawProfiles)
+	actualProfiles := c.clientWrapper.UserProfileService.(userprofileservice.UPSHelper).GetUserProfiles()
+
+	success := false
+	for _, expectedProfile := range expectedProfiles {
+		success = false
+		for index, actualProfile := range actualProfiles {
+			if subset.Check(expectedProfile, actualProfile) {
+				// Removing already found profile from actual profiles for performance
+				actualProfiles = append(actualProfiles[:index], actualProfiles[index+1:]...)
+				success = true
+				break
+			}
+		}
+		if !success {
+			break
+		}
+	}
+	if success {
+		return nil
+	}
+
+	return fmt.Errorf("User profile state not equal")
```

Comment:
We should log the different found between the user profiles

Reply:
done 👍